src/component/TodoItem.js

```diff
@@ -0,0 +1,75 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import Checkbox from './checkboxInput';
+
+const TodoItem = ({
+  todo, checkboxHandler, removeHandler, setUpdate,
+}) => {
+  const [state, setState] = useState({
+    editing: false,
+  });
+  const completedStyle = {
+    fontStyle: 'italic',
+    color: '#595959',
+    opacity: 0.4,
+    textDecoration: 'line-through',
+  };
+
+  const handleEditing = () => {
+    setState({
+      editing: true,
+    });
+  };
+
+  const viewMode = {};
+  const editMode = {};
+
+  if (state.editing) {
+    viewMode.display = 'none';
+  } else {
+    editMode.display = 'none';
+  }
+
+  const handleUpdatedDone = (e) => {
+    if (e.key === 'Enter') {
+      setState({ editing: false });
+    }
+  };
+
+  useEffect(() => () => {
+    console.log('cleaning up...');
```

Comment:
Please remove this console.log because it's not very clean. It's also not good practice to use them because when forgotten, critical data could be accidentally displayed to users in production.